src/apigateway/mitol/apigateway/views.py

```diff
@@ -0,0 +1,87 @@
+"""Custom logout view for the API Gateway."""
+
+import logging
+
+from django.conf import settings
+from django.contrib.auth import logout
+from django.shortcuts import redirect
+from django.utils.http import url_has_allowed_host_and_scheme
+from django.views import View
+
+log = logging.getLogger(__name__)
+
+
+def get_redirect_url(request):
+    """
+    Get the redirect URL from the request.
+
+    Args:
+        request: Django request object
+
+    Returns:
+        str: Redirect URL
+    """
+    log.debug("views.get_redirect_url: Request GET is: %s", request.GET.get("next"))
+    log.debug(
+        "views.get_redirect_url: Request cookie is: %s", request.COOKIES.get("next")
+    )
+
+    next_url = request.GET.get("next") or request.COOKIES.get("next")
+    log.debug("views.get_redirect_url: Redirect URL (before valid check): %s", next_url)
+
+    if request.COOKIES.get("next"):
+        # Clear the cookie after using it
+        log.debug("views.get_redirect_url: Popping the next cookie")
+
+        request.COOKIES.pop("next", None)
+
+    return (
+        next_url
+        if next_url
+        and url_has_allowed_host_and_scheme(
+            next_url, allowed_hosts=settings.ALLOWED_HOSTS
```

Comment:
Should this be `ALLOWED_REDIRECT_HOSTS`?  I think `ALLOWED_HOSTS` has a different purpose - https://docs.djangoproject.com/en/5.1/ref/settings/#allowed-hosts